src/chemistry/network/kida.hpp

```diff
@@ -0,0 +1,226 @@
+  int nices_;//number of ice species
+
+  //physical quantities
+  const Real xHe_ = 0.1;//Helium abundance
+  //whether to cap temperature if the reaction is outside of the temperature range
+  //only for 2body reactions. Default is false, which means extrapolation
+  bool is_Tcap_2body_;
+  //whether to fix dust metallicity for implicit grain assisted
+  //reactions (opposed to reading from the initial grain abundance from the
+  //input file)
+  bool is_fixed_Zd_;
+  //whether to fix PAH abundance for PE heating and Rec cooling
+  //if so, use PE heating and Rec cooling from Draine & Weingartner
+  //if not, use PE heating and Rec cooling from Wolfire+ 2003
+  bool is_fixed_PAH_;
+  //flag for in isothermal EOS only temperature related rates are
+  //calculated only once in the beginning
+  bool flag_T_rates_;
+  Real Z_g_; //gas metallicity relative to solar, default 1.
+  Real Z_d_; //larger dust grain metallicity relative to solar, default 1.
+  Real Z_PAH_; //PAH abundance, for PE heating and Rec cooling
+  Real phi_PAH_; //PAH recombination efficiency, default 0.4
+  Real a_d_; //size of the dust grain in cm, default 1e-5 (0.1 micron)
+  Real rho_d_; //density of grain in cgs, default 2 g/cm3
+  Real m_d_; //mass of the dust grain in g
+  Real x_d_; //relative abundance of all dust
+  Real nH_; //density, updated at InitializeNextStep from hydro variable
+  Real o2pH2_;//ortho to para H2 ratio, default 3:1
+  Real Yi_;//Yield for crp desorption, default 1e-3
+  Real temperature_; //temperature of the gas if isothermal
+  Real temp_min_rates_; //temperature floor for reaction rates
+  Real temp_min_cool_; //temperature minimum for cooling
+  Real temp_max_cool_nm_; //temperature maximum for neutral medium cooling
+  Real temp_dust_thermo_; //dust temperature for dust thermo cooling and desorption
+
+  //reaction constants
+  //special rates index map
+  int id7max_;
+  AthenaArray<int> id7map_;
+  AthenaArray<ReactionType> id7type_;
+  //direct cosmic-ray ionization
+  int n_cr_;
+  AthenaArray<int> incr_;
+  AthenaArray<int> outcr1_;
+  AthenaArray<int> outcr2_;
+  AthenaArray<int> outcr3_;
+  AthenaArray<Real> kcr_base_;
+  AthenaArray<Real> kcr_;
+  //index for cr ionization rates for CR heating
+  int icr_H_;
+  int icr_H2_;
+  int icr_He_;
+  //cosmic-ray induced photo ionization
+  int n_crp_;
+  AthenaArray<int> incrp_;
+  AthenaArray<int> outcrp1_;
+  AthenaArray<int> outcrp2_;
+  AthenaArray<Real> kcrp_base_;
+  AthenaArray<Real> kcrp_;
+  //photo reactions
+  int n_ph_;
+  AthenaArray<int> inph_;
+  AthenaArray<int> outph1_;
+  AthenaArray<int> outph2_;
+  //reactant species name map, for radiation calculation
+  std::map<std::string, int> smap_ph_;
+  AthenaArray<Real> kph_base_;
+  AthenaArray<Real> kph_avfac_;
+  AthenaArray<Real> kph_;
+  //index for H2 photodissociation for H2 UV pumping and dissociation heating
+  int iph_H2_;
+  //2body reactions
+  int n_2body_;
+  const int n_in2body_ = 2;
```

Comment:
can these and others in the file be `constexpr`?

Reply:
Yes!